### PR TITLE
[DYN-6868] migrate to typeScript

### PR DESCRIPTION
**Pull request by Enzo707**
This PR is for migrating DynamoHome from js to typeScript as a first step before adding test coverage.
Ref.: [DYN-6868](https://jira.autodesk.com/browse/DYN-6868)

**Review**
@QilongTang 
@dnenov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,16 +3,16 @@
   "version": "1.0.15",
```
Comment (QilongTang) [inline package.json:3]: please bump up version always

**Comment by QilongTang** [comment]
@Enzo707 Did you get the chance to fix the unit tests?

File: src/components/LayoutContainer.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
-import React, { useState, useEffect } from 'react';
-import { MainContent } from './MainContent.jsx';
-import { Sidebar } from './Sidebar/Sidebar.jsx';
+import { useState } from 'react';
+import { MainContent } from '../components/MainContent';
+import { Sidebar } from '../components/Sidebar/Sidebar';
```
Comment (QilongTang) [inline src/components/LayoutContainer.tsx:3]: Cant we import from the same folder?

File: src/components/MainContent.tsx [hunk 3]
```diff
@@ -1,27 +1,27 @@
-import React, { useState, useEffect } from 'react';
-import { RecentPage } from './Recent/PageRecent.jsx';
-import { SamplesPage } from './Samples/PageSamples.jsx';
-import { LearningPage } from './Learning/PageLearning.jsx';
+import { useState, useEffect } from 'react';
+import { RecentPage } from '../components/Recent/PageRecent';
+import { SamplesPage } from '../components/Samples/PageSamples';
+import { LearningPage } from '../components/Learning/PageLearning';
```
Comment (QilongTang) [inline src/components/MainContent.tsx:4]: same question here

File: src/components/Recent/PageRecent.tsx [hunk 4]
```diff
@@ -1,27 +1,27 @@
...
 import { useState, useEffect } from 'react';
-import { GraphGridItem } from './GraphGridItem.jsx';
-import { CustomNameCellRenderer } from './CustomNameCellRenderer.jsx';
-import { CustomLocationCellRenderer } from './CustomLocationCellRenderer.jsx';
-import { CustomAuthorCellRenderer } from "./CustomAuthorCellRenderer.jsx";
-import { GraphTable } from './GraphTable.jsx';
-import { GridViewIcon, ListViewIcon } from '../Common/CustomIcons.jsx';
-import { openFile, saveHomePageSettings } from '../../functions/utility.js';
+import { GraphGridItem } from './GraphGridItem';
+import { CustomNameCellRenderer } from './CustomNameCellRenderer';
```
Comment (QilongTang) [inline src/components/Recent/PageRecent.tsx:4]: We are able to import from the folder here?

**Review by QilongTang — COMMENTED** [review]
LGTM with couple questions

File: src/components/LayoutContainer.tsx (hunk 2 above)
Comment (Enzo707) [reply src/components/LayoutContainer.tsx:3]: Sure, we can also implement an index file and import everything from feature folder or just from /components root:

`import { MainContent } from './MainContent';
import { Sidebar } from './Sidebar';`
Or
`import { MainContent, Sidebar } from './';`

Comment (QilongTang) [reply src/components/LayoutContainer.tsx:3]: Thanks, @Enzo707 Would you update? I would like the import statements to be consistent

Comment (Enzo707) [reply src/components/LayoutContainer.tsx]: @QilongTang I already updated the imports. PTAL

**Review by QilongTang — APPROVED** [review]

**Pull request merged**